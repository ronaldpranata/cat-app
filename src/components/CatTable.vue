<template>
  <div class="table">
    <div class="row header">
      <div class="column" v-for="(item, index) in headers" :key="index">
        {{ item }}
      </div>
    </div>
    <div
      class="row body clickable"
      v-for="(item, index) in body"
      :key="index"
      :id="item?.name"
      v-on:click="gotoDetail(item?.name)"
    >
      <div class="column" v-for="(value, key) in item" :key="key">
        <span v-if="key == 'image'" :class="key" >
          <img :src="value" width="60" />
        </span>
        <span v-else :class="key"  >
          {{ value }}
        </span>
      </div>
    </div>
  </div>
</template>

<script>
import { defineComponent } from 'vue';
import { useI18n } from 'vue-i18n';

export default defineComponent({
  name: 'CatTable',
  props: {
    headers: Array,
    body: Array,
  },
  setup() {
    const { t } = useI18n();
    return { t };
  },
  methods: {
    gotoDetail(name) {
      this.$emit('gotoDetail', name);
    },
  },
});
</script>
<style scoped lang='scss'>
.table {
  max-width: 500px;
  margin: auto;

  .row {
    display: flex;
    flex-direction: row;
    padding: 10px;

    &.header {
      font-weight: bold;
      padding: 10px;
    }
  }

  .column {
    display: flex;
    flex: 1;
    flex-direction: column;
  }
}
</style>
