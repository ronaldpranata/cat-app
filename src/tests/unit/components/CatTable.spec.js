import { shallowMount } from '@vue/test-utils';
import { createI18n } from 'vue-i18n';
import CatTable from '@/components/CatTable.vue';

const i18n = createI18n({
  // vue-i18n options here ...
});
const factory = () => shallowMount(CatTable, {
  propsData: {
    header: ['test', 'test'],
    body:
      [
        {
          image: 'image',
          name: 'cat1',
        },
        {
          image: 'image2',
          name: 'cat2',
        },
      ],
  },
  global: {
    plugins: [i18n],
  },
});

describe('CatTable', () => {
  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
