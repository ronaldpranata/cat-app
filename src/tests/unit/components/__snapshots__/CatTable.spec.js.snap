// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatTable renders properly 1`] = `
<div class="table" header="test,test">
  <div class="row header"></div>
  <div class="row body clickable">
    <div class="column"><span><img src="image" width="60"></span></div>
    <div class="column"><span>cat1</span></div>
  </div>
  <div class="row body clickable">
    <div class="column"><span><img src="image2" width="60"></span></div>
    <div class="column"><span>cat2</span></div>
  </div>
</div>
`;
